presidio-analyzer/analyzer/recognizer_registry/recognizer_registry.py

```diff
@@ -0,0 +1,96 @@
+import logging
+
+from analyzer import PatternRecognizer
+from analyzer.predefined_recognizers import CreditCardRecognizer, \
+    SpacyRecognizer, CryptoRecognizer, DomainRecognizer, \
+    EmailRecognizer, IbanRecognizer, IpRecognizer, NhsRecognizer, \
+    UsBankRecognizer, UsLicenseRecognizer, \
+    UsItinRecognizer, UsPassportRecognizer, UsPhoneRecognizer, UsSsnRecognizer
+
+
+class RecognizerRegistry:
+    """
+    Detects, registers and holds all recognizers to be used by the analyzer
+    """
+
+    def __init__(self, recognizers=None):
+        if recognizers is None:
+            recognizers = []
+        self.recognizers = recognizers
+
+    def load_recognizers(self, path):
+        #   TODO: Change the code to dynamic loading -
+        # Task #598:  Support dynamic loading of the pre-defined recognizers
+        self.recognizers.extend([CreditCardRecognizer(),
+                                 SpacyRecognizer(),
+                                 CryptoRecognizer(), DomainRecognizer(),
+                                 EmailRecognizer(), IbanRecognizer(),
+                                 IpRecognizer(), NhsRecognizer(),
+                                 UsBankRecognizer(), UsLicenseRecognizer(),
+                                 UsItinRecognizer(), UsPassportRecognizer(),
+                                 UsPhoneRecognizer(), UsSsnRecognizer()])
+
+    def add_pattern_recognizer_from_dict(self, recognizer_dict):
+        """
+        Creates a pattern recognizer from a dictionary
+         and adds it to the recognizers list
+        :param recognizer_dict: A pattern recognizer serialized
+         into a dictionary
+        """
+
+        pattern_recognizer = PatternRecognizer.from_dict(recognizer_dict)
+
+        for rec in self.recognizers:
+            if rec.name == pattern_recognizer.name:
+                raise ValueError(
+                    "Recognizer of name {} is already defined".format(
+                        rec.name))
+
+        self.recognizers.append(pattern_recognizer)
+
+    def remove_recognizer(self, name):
+        found = False
+        for index, rec in enumerate(self.recognizers):
+            if rec.name == name:
+                found = True
+                self.recognizers.pop(index)
+
+        if not found:
+            raise ValueError("Requested recognizer was not found")
+
+    def get_recognizers(self, entities=None, language=None):
+        if language is None and entities is None:
+            return self.recognizers
+
+        if language is None:
```

Comment:
We had a discussion around this. For now we don't support multi-language recognizers, mainly because we can never know which language the user would ask and whether the recognizer would actually support it (for example, a credit card number recognizer which is given a text in arabic or chinese)